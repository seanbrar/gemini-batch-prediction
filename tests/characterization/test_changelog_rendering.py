"""
Characterization test for Jinja2 changelog template rendering.
"""

import pytest


@pytest.mark.golden_test("golden_files/test_changelog_rendering.yml")
def test_changelog_template_renders_correctly(
    golden, jinja_env, mock_changelog_context
):
    """
    Characterizes the full output of the changelog Jinja template.

    This test avoids running the semantic-release CLI. Instead, it mocks
    the context data that is passed to the template and calls the rendering
    engine directly. This provides a fast, reliable, and deterministic test
    that fully covers the template's logic, including:
    - Section generation (features, fixes, etc.)
    - Commit formatting (scope, hash, links)
    - Breaking change sections
    - Released and Unreleased versions
    """
    # Arrange
    template = jinja_env.get_template("CHANGELOG.md.j2")

    # Act
    # Render the template using our mock context
    actual_changelog = template.render(context=mock_changelog_context)

    # Assert
    # The golden file contains the complete, expected changelog string
    assert actual_changelog == golden.out["expected_changelog"]
