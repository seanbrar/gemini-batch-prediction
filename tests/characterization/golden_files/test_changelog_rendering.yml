# This file captures the expected output of the changelog Jinja template
# Includes commits with no scope, with linked issues, and duplicate
# breaking changes to provide a more robust characterization test.
expected_changelog: |
  # Changelog

  All notable changes to this project will be documented in this file.

  The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
  and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

  ---

  <!-- PSR-INSERT-FLAG -->
  ## [Unreleased]

  ### üí• Breaking Changes

  - A breaking change related to rework authentication flow.

  ### ‚ú® Features

  - **(api)** Add exciting new endpoint ([`f34t001`](https://github.com/USER/REPO/commit/f34t001f34t001f34t001f34t001f34t001))
  - **(client)** Link to external issues ([`f34t002`](https://github.com/USER/REPO/commit/f34t002f34t002f34t002f34t002f34t002)) ([#123](https://github.com/USER/REPO/issues/123), [#456](https://github.com/USER/REPO/issues/456))

  ### üêõ Bug Fixes

  - Correct a minor typo in the README ([`f18x002`](https://github.com/USER/REPO/commit/f18x002f18x002f18x002f18x002f18x002))
  - **(client)** Resolve critical connection bug ([`f18x001`](https://github.com/USER/REPO/commit/f18x001f18x001f18x001f18x001f18x001))

  ### ‚ôªÔ∏è Refactoring

  - **(auth)** Rework authentication flow ([`r3f4c71`](https://github.com/USER/REPO/commit/r3f4c71r3f4c71r3f4c71r3f4c71r3f4c71))
  - **(core)** Simplify internal logic ([`r3f4c70`](https://github.com/USER/REPO/commit/r3f4c70r3f4c70r3f4c70r3f4c70r3f4c70))

  ---

  ## [1.0.0] - 2024-07-01

  ### üí• Breaking Changes

  - A breaking change related to implement initial logic.

  ### ‚ú® Features

  - **(core)** Implement initial logic ([`b4dfeat`](https://github.com/USER/REPO/commit/b4dfeatb4dfeatb4dfeatb4dfeatb4dfeat))

  ### üìö Documentation

  - Add README and contributing guide ([`d0c5001`](https://github.com/USER/REPO/commit/d0c5001d0c5001d0c5001d0c5001d0c5001))

  ### üì¶ Build System & Dependencies

  - Configure pyproject.toml ([`b51ld01`](https://github.com/USER/REPO/commit/b51ld01b51ld01b51ld01b51ld01b51ld01))

  ---

  <!-- PSR-LINKS-START -->
  [Unreleased]: https://github.com/USER/REPO/compare/v1.0.0...HEAD
  [1.0.0]: https://github.com/USER/REPO/releases/tag/v1.0.0
